src/Microsoft.Azure.WebJobs.Host/Config/FluentBindingRule.cs

```diff
@@ -106,6 +108,43 @@ public FluentBindingRule<TAttribute> WhenIsNotNull(string propertyName)
             return this;
         }
 
+        /// <summary>
+        /// The subsequent Bind* operations only apply when the Attribute's property is not null. 
+        /// </summary>
+        /// <param name="propertyName"></param>
+        /// <returns></returns>
+        public FluentBindingRule<TAttribute> When<TValue>(string propertyName, TValue expected)
```

Comment:
>FluentBindingRule [](start = 15, length = 17)

Blob uses this  (only supports CloudBlobStream on Access=Write). Graph binding (or anything else with a "mode") will benefit from it too.